csharp/src/Google.Protobuf/MessageParser.cs

```diff
@@ -65,7 +68,7 @@ internal IMessage CreateTemplate()
         public IMessage ParseFrom(byte[] data)
         {
             IMessage message = factory();
-            message.MergeFrom(data);
+            message.MergeFrom(data, DiscardUnknownFields);
```

Comment:
Actually I want to ask if these changes are tested. The comments say it is "A general message parser, typically used by reflection-based code as all the methods" but I can't find tests for the reflection based parse. I guess the tests you added can also be parse without change this general message parser?

Reply:
The tests I've provided call all the parse methods, because they use parsers declared as `MessageParser<T>` and `MessageParser`.